herbert_core/classes/MPIFramework/parallel_worker.m

```diff
@@ -295,7 +298,9 @@
     fbMPI = je.control_node_exch;
     % is framework instance different from JE instance now?
     intercomm= je.mess_framework;
-    
+    % migrate job folder for message exchange without deleting the old
+    % one
+    je.migrate_job_folder(false);
```

Comment:
why is the folder migrated, but not cleaned up in this use-case?

Reply:
the workers and the server itself check the presence of the exchange folder and if it is deleted, fail. To ensure this would not happen, for workers during migration, I have done changes to JobExecutor.finish_task. But server still checks the folder existence and if not find one, fails itself. This is why I've dedicated the honor of deleting job folder to server only

Reply:
the migration not only for failures. The normal job may still leave some garbage that will confuse the next job and may have issues with startup files cleared up by the previous job (hopefully resolved earlier)

Reply:
In addition to that, the server may not pick up a client outputs, so clients should not delete exchange folder